File: examples/dialog/src/main/java/DialogExample.java
import io.github.kolod.FontDialog;

import javax.swing.*;
import java.awt.Frame;
import java.lang.reflect.InvocationTargetException;

public class DialogExample implements Runnable {

    public static void main(String[] args) throws InvocationTargetException, InterruptedException {
        ExampleRunner.useLookAndFeel(UIManager.getSystemLookAndFeelClassName());
        ExampleRunner.invoke(new DialogExample());
    }

    @Override
    public void run() {
        FontDialog dialog = new FontDialog((Frame) null, "Select Font", true);
        dialog.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
        dialog.setVisible(true);
        if (!dialog.isCancelSelected()) {
            System.out.printf("Selected font is: %s%n", dialog.getSelectedFont());
        }
    }

}

File: examples/dialog/src/main/kotlin/DialogExample.kt
package io.github.kolod

import java.awt.Frame
import java.lang.reflect.InvocationTargetException
import javax.swing.UIManager
import javax.swing.WindowConstants

class DialogExample : Runnable {

    override fun run() {
        val dialog = FontDialog(null as Frame?, "Select Font", true)
        dialog.defaultCloseOperation = WindowConstants.DISPOSE_ON_CLOSE
        dialog.setLocationRelativeTo(null)
        dialog.isVisible = true
        if (!dialog.isCancelSelected) {
            System.out.printf("Selected font is: %s%n", dialog.selectedFont)
        }
    }

    companion object {
        @Throws(InvocationTargetException::class, InterruptedException::class)
        @JvmStatic
        fun main(args: Array<String>) {
            ExampleRunner.useLookAndFeel(UIManager.getSystemLookAndFeelClassName())
            ExampleRunner.invoke(DialogExample())
        }
    }
}
